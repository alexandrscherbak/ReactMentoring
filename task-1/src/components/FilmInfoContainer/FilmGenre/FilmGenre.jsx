import React from 'react';

const FilmGenre = ({genre}) => (
	<div>
		<span>{genre}</span>

		<style jsx>
			{`
				span {
					font-size: 12px;
					color: #9e9e9e;
				}
			`}
		</style>
	</div>
);

export default FilmGenre;
