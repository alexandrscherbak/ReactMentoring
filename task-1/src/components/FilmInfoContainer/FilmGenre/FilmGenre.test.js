import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmGenre from './FilmGenre.jsx';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const props = {
	genre: 'action',
};

function setup() {
	return mount(<FilmGenre {...props}/>);
}

describe('FilmGenre', () => {
	test('snapshot', () => {
		const component = renderer.create(
			<FilmGenre {...props}/>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('should render self and subcomponents', () => {
		const enzymeWrapper = setup();
		expect(enzymeWrapper.find('span').text()).toBe(props.genre);
	});
});
