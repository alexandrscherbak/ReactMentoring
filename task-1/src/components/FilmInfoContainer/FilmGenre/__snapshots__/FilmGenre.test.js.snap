// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmGenre snapshot 1`] = `
<div
  className="jsx-672986837"
>
  <span
    className="jsx-672986837"
  >
    action
  </span>
</div>
`;
